core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt

```diff
@@ -78,31 +83,22 @@ internal class CBPeripheralCoreBluetoothPeripheral(
 
     private val scope = CoroutineScope(
         parentCoroutineContext +
-            SupervisorJob(parentCoroutineContext.job) +
+            SupervisorJob(parentCoroutineContext.job).apply { invokeOnCompletion(::dispose) } +
             CoroutineName("Kable/Peripheral/${cbPeripheral.identifier.UUIDString}"),
     )
 
-    private val centralManager: CentralManager = CentralManager.Default
-
     private val logger = Logger(logging, identifier = cbPeripheral.identifier.UUIDString)
 
-    private val _state = MutableStateFlow<State>(State.Disconnected())
+    private val centralManager: CentralManager = CentralManager.Default
+
+    private val _state = MutableStateFlow<State>(Disconnected())
     override val state: StateFlow<State> = _state.asStateFlow()
 
     override val identifier: Uuid = cbPeripheral.identifier.toUuid()
 
     private val observers = Observers<NSData>(this, logging, exceptionHandler = observationExceptionHandler)
 
     init {
-        centralManager.delegate
-            .state
-            .filter { state -> state == CBManagerStatePoweredOff }
-            .onEach {
-                disconnect()
-                _state.value = State.Disconnected()
-            }
-            .launchIn(scope)
```

Comment:
This was moved within `establishConnection`, so it will now match the lifecycle of a connection rather than living for the entire life of the peripheral.

This is a step in the process of resolving #378.